src/components/Calculator.js

```diff
@@ -1,33 +1,64 @@
 import React from 'react';
 import './Calculator.css';
+import calculate from '../logic/calculate';
 
-function Calculator() {
-  return (
-    <main>
-      <div className="calculator">
-        <div className="display">0</div>
-        <div className="key">AC</div>
-        <div className="key">+/-</div>
-        <div className="key">%</div>
-        <div className="key orange">&divide;</div>
-        <div className="key">9</div>
-        <div className="key">8</div>
-        <div className="key">7</div>
-        <div className="key orange">&times;</div>
-        <div className="key">6</div>
-        <div className="key">5</div>
-        <div className="key">4</div>
-        <div className="key orange">-</div>
-        <div className="key">3</div>
-        <div className="key zero">2</div>
-        <div className="key">1</div>
-        <div className="key orange">÷</div>
-        <div id="zero" className="key">0</div>
-        <div className="key">.</div>
-        <div className="key orange">=</div>
-      </div>
-    </main>
-  );
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      total: 0,
+      next: '',
+      operation: '',
+    };
+    this.onClickHandler = this.onClickHandler.bind(this);
+  }
+
+  onClickHandler(e) {
+    const targetValue = e.target.value;
+    this.setState((state) => calculate({
+      next: state.next,
+      total: state.total,
+      operation: state.operation,
+    }, targetValue));
+  }
+
+  render() {
+    const { total, operation, next } = this.state;
+    return (
+      <main>
+        <div className="calculator">
+          <div className="display">
+            {total}
+            {' '}
+            {operation}
+            {' '}
+            {next}
+            {' '}
+          </div>
+          <button onClick={this.onClickHandler} value="AC" className="key" type="button">AC</button>
+          <button onClick={this.onClickHandler} value="%" className="key" type="button">%</button>
+          <button onClick={this.onClickHandler} value="+/-" className="key" type="button">+/-</button>
+          <button onClick={this.onClickHandler} value="&divide;" className="key orange" type="button">&divide;</button>
+          <button onClick={this.onClickHandler} value="9" className="key" type="button">9</button>
+          <button onClick={this.onClickHandler} value="8" className="key" type="button">8</button>
+          <button onClick={this.onClickHandler} value="7" className="key" type="button">7</button>
+          <button onClick={this.onClickHandler} value="&times;" className="key orange" type="button">&times;</button>
```

Comment:
- [ ] When the user tries to multiply two numbers, clicking on the `=` sign redirects to a blank page. Kindly fix this issue

Reply:
- Well resolved 👍🏽